filesystem/node.go

```diff
@@ -157,34 +159,21 @@ func (n *Node) Setattr(_ context.Context, req *fuse.SetattrRequest, resp *fuse.S
 		return nil
 	}
 
-	// Wrap everything inside a transaction.
-	tx, err := n.cfs.db.Begin()
-	if err != nil {
-		log.Print(err)
-		return err
-	}
+	// Store the current size in case we need to rollback.
+	originalSize := n.Size
```

Comment:
this is the important bit, it needs to be outside the transaction function. Ditto for the second block of changes in this file.